### DISCUSSION TITLE
How would a _data filename with dash (`-`) in it would be accessible from nunjucks template?

### DISCUSSION THREAD

**Discussion by amithm7** [11ty/eleventy#2620] [Q&A]
```
_data
  file-name.json
index.njk
```

In `index,njk` how would I access `file-name` ?

Tried these:

```njk
{{ file-name }}

{{ 'file-name' }}

{{ ['file-name'] }}
```

**Comment by pdehaan** [comment]
Possibly related https://github.com/11ty/eleventy/issues/567

When possible, I use underscores instead of dashes, or even put the data file in a subdirectory.

**Reply by pdehaan** [reply]
You might also be able to do some `eleventyComputed` magic to remap the global data file name to be more friendly:
```njk
---js
{
  title: "HoMePaGe",
  eleventyComputed: {
    fileName: (data) => data['file-name']
  }
}
---

<h1>{{ title }}</h1>

<pre>
{{ fileName.name }}
</pre>
```

**Reply by amithm7** [reply]
> When possible, I use underscores instead of dashes, or even put the data file in a subdirectory.

Yes, I am just used to naming files as `some-file-name.extension`. I did rename the file I had problem with to camelCase.

`eleventyComputed` magic doesn't help much if there are many file names with dash.

https://github.com/11ty/eleventy/issues/567#issuecomment-575828788 , looks like a better solution.

Thanks.
